### PR TITLE
{lyn14000} re-insert script processing rule logic

### PR DESCRIPTION
**Pull request by jackalbe**
## What does this PR do?

This stores the "last used" Python script in the ScriptProcessorRuleBehavior so that when the manifest is updated then the next script events such as OnPrepareForExport can be executed latter on. 

The python script stack gets cleared out between scenes being processed.

## How was this PR tested?

New auto test:
```
test_ProcessAssetScript_SimpleFbx_HasExpectedProducts[windows-AutomatedTesting] PASSED        [100%]
```

### CODE REVIEW

**Review by amzn-mike — APPROVED** [review]

**Review by AMZN-stankowi — APPROVED** [review]

File: AutomatedTesting/Assets/ap_test_assets/script_reinsert.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+try:
+    import azlmbr.scene as sceneApi
+    
+    if sceneJobHandler is None:
+        sceneJobHandler = sceneApi.ScriptBuildingNotificationBusHandler()
+        sceneJobHandler.connect()
+        sceneJobHandler.add_callback('OnUpdateManifest', on_update_manifest)
+        sceneJobHandler.add_callback('OnPrepareForExport', on_prepare_for_export)
+
+except:
```
Comment (junbo75) [inline AutomatedTesting/Assets/ap_test_assets/script_reinsert.py:77]: Nit: Is there any specific type of exception we're trying to catch here?

File: AutomatedTesting/Gem/PythonTests/assetpipeline/fbx_tests/assets/script_basics/base_example.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+try:
+    import azlmbr.scene as sceneApi
+    
+    if sceneJobHandler is None:
+        sceneJobHandler = sceneApi.ScriptBuildingNotificationBusHandler()
+        sceneJobHandler.connect()
+        sceneJobHandler.add_callback('OnUpdateManifest', on_update_manifest)
+        sceneJobHandler.add_callback('OnPrepareForExport', on_prepare_for_export)
+
+except:
```
Comment (junbo75) [inline AutomatedTesting/Gem/PythonTests/assetpipeline/fbx_tests/assets/script_basics/base_example.py:77]: Nit: Same here. Is there any specific type of exception we're trying to catch?

**Review by junbo75 — APPROVED** [review]

File: AutomatedTesting/Assets/ap_test_assets/script_reinsert.py (hunk 1 above)
Comment (jackalbe) [reply AutomatedTesting/Assets/ap_test_assets/script_reinsert.py:77]: not really, any exceptions will fail the test script which will fail the build

File: AutomatedTesting/Gem/PythonTests/assetpipeline/fbx_tests/assets/script_basics/base_example.py (hunk 2 above)
Comment (jackalbe) [reply AutomatedTesting/Gem/PythonTests/assetpipeline/fbx_tests/assets/script_basics/base_example.py:77]: not really, any exceptions will fail the test script which will fail the build

**Pull request merged**